File: 3773-minimum-pair-removal-to-sort-array-i/solution.java
class Solution {
    public int minimumPairRemoval(int[] nums) {
        int n = nums.length;
        List<Integer> ans = new ArrayList<>();
        for(int i = 0; i<nums.length; i++){
            ans.add(nums[i]);
        }
        int ops = 0;
        while(!isSorted(ans)) {
            int minSum = Integer.MAX_VALUE;
            int index = -1;
            for(int i = 0; i<ans.size()-1; i++){
                int sum = ans.get(i) + ans.get(i+1);
                if(sum<minSum){
                    minSum = sum;
                    index = i;
                }
            }
            int mergevalue = ans.get(index) + ans.get(index+1);
            ans.set(index , mergevalue);
            ans.remove(index+1);
            ops++;

        }
        return ops; 
        
    }
    private boolean isSorted(List<Integer>ans){
        for(int i = 1; i<ans.size(); i++){
            if(ans.get(i)<ans.get(i-1)){
                return false;
            }
        }
        return true;
    }
}

File: 3773-minimum-pair-removal-to-sort-array-i/solution.cpp
class Solution {
public:
    int minimumPairRemoval(vector<int>& nums) {
        int n = nums.size();
        vector<int> ans;

        // copy nums into ans
        for (int i = 0; i < nums.size(); i++) {
            ans.push_back(nums[i]);
        }

        int ops = 0;

        while (!isSorted(ans)) {
            int minSum = INT_MAX;
            int index = -1;

            // find adjacent pair with minimum sum
            for (int i = 0; i < ans.size() - 1; i++) {
                int sum = ans[i] + ans[i + 1];
                if (sum < minSum) {
                    minSum = sum;
                    index = i;
                }
            }

            // merge the pair
            int mergevalue = ans[index] + ans[index + 1];
            ans[index] = mergevalue;
            ans.erase(ans.begin() + index + 1);
            ops++;
        }

        return ops;
    }

private:
    bool isSorted(vector<int>& ans) {
        for (int i = 1; i < ans.size(); i++) {
            if (ans[i] < ans[i - 1]) {
                return false;
            }
        }
        return true;
    }
};

